http/server_test.go

```diff
@@ -15,36 +14,30 @@ import (
 	"time"
 
 	"github.com/packethost/pkg/log"
+	"github.com/stretchr/testify/require"
 	"github.com/tinkerbell/hegel/datamodel"
-	grpcserver "github.com/tinkerbell/hegel/grpc-server"
+	"github.com/tinkerbell/hegel/grpc"
 	"github.com/tinkerbell/hegel/hardware"
 	"github.com/tinkerbell/hegel/hardware/mock"
-	"github.com/tinkerbell/hegel/metrics"
+	_ "github.com/tinkerbell/hegel/metrics" // Initialize metrics.
 	"github.com/tinkerbell/hegel/xff"
 )
 
-func TestMain(m *testing.M) {
-	l, _ := log.Init("github.com/tinkerbell/hegel")
-	logger = l.Package("httpserver")
-	metrics.Init(l)
-
-	hc := mock.HardwareClient{}
-	hegelServer = grpcserver.NewServer(logger, hc)
-
-	os.Exit(m.Run())
-}
-
 // TestTrustedProxies tests if the actual remote user IP is extracted correctly from the X-FORWARDED-FOR header according to the list of trusted proxies provided.
 func TestTrustedProxies(t *testing.T) {
+	logger, err := log.Init(t.Name())
+	require.NoError(t, err)
```

Comment:
This and elsewhere can be replaced with

```suggestion
	logger := log.Test(t, t.Name())
```